_data-prepper/common-use-cases/codec-processor-combinations.md

```diff
@@ -0,0 +1,55 @@
+---
+layout: default
+title: Codec processor combinations
+parent: Common use cases
+nav_order: 25
+---
+
+# Codec processor combinations
+
+At ingestion time, data received by the [`s3` source]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/sources/s3/) can be parsed by [codecs]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/sources/s3#codec). which will compress and decompress large sets of data in a certain format before ingestion through a Data Prepper pipelines [processor]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/processors/).
+
+While most codecs can be used with most processors, the following codec processor combinations can help make your pipeline more efficient when using the following input types.
+
+## JSON Array
+
+A [JSON array](https://json-schema.org/understanding-json-schema/reference/array) is used to order elements of different types. Because an array is required in JSON, the data contained within the array must be tabular.
+
+The JSON array does not require a processor. 
+
+## NDJSON
+
+Unlike a JSON array, [NDJSON](https://www.npmjs.com/package/ndjson), each row of data can be delimited by a newline, which means data is processed per-line as opposed to an array.
```

Comment:
```suggestion
Unlike a JSON array, [NDJSON](https://www.npmjs.com/package/ndjson) allows for each row of data to be delimited by a newline, meaning data is processed per line instead of an array.
```